client/js/store/procedure/SplitStatementStore.js

```diff
@@ -230,8 +230,8 @@ const SplitStatementStore = {
               for (let j = i + 1; j < segments.length; j++) {
                 // Check for overlap
                 if (
-                  (segments[i].charStart >= segments[j].charStart && segments[i].charStart <= segments[j].charEnd) ||
-                  (segments[j].charStart >= segments[i].charStart && segments[j].charStart <= segments[i].charEnd)
+                  (segments[i].charStart > segments[j].charStart && segments[i].charStart < segments[j].charEnd) ||
```

Comment:
Curious question: Why do we loop twice here?

Reply:
Because we want to compare all segments in case they are not ordered. Without a nested loop you can only compare one segment with the next one.